### PR TITLE
added emotion analyzer module and attched in the open ai api

### PR DESCRIPTION
**Pull request by soulsyrup**

### CODE REVIEW

File: fastchat/modules/emotion_analyzer.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+def get_cached_analysis(text: str):
+    cached_result = r.get(text)
+    if cached_result:
+        return json.loads(cached_result)
+    else:
+        analysis_result = emotion_analyzer_function(text)
+        r.set(text, json.dumps(analysis_result))
+        return analysis_result
+
+@app.post("/emotion_analyze")
```
Comment (digisomni) [inline fastchat/modules/emotion_analyzer.py]: Should this be here?

File: fastchat/serve/openai_api_server.py [hunk 2]
```diff
@@ -662,6 +662,10 @@ async def get_embedding(payload: Dict[str, Any]):
 
 ### GENERAL API - NOT OPENAI COMPATIBLE ###
 
+@app.post("/emotion_analyze")
```
Comment (digisomni) [inline fastchat/serve/openai_api_server.py]: Does the emotion analyze module need to be imported? Also this should start with /api/v1/

File: fastchat/serve/openai_api_server.py [hunk 3]
```diff
@@ -662,6 +662,10 @@ async def get_embedding(payload: Dict[str, Any]):
...
+def analyze_text(item: Text):
```
Comment (digisomni) [inline fastchat/serve/openai_api_server.py]: How is this linking back to the module?

**Review by digisomni — CHANGES_REQUESTED** [review]

File: fastchat/modules/emotion_analyzer.py (hunk 1 above)
Comment (soulsyrup) [reply fastchat/modules/emotion_analyzer.py]: no, I removed it now

File: fastchat/serve/openai_api_server.py (hunk 3 above)
Comment (soulsyrup) [reply fastchat/serve/openai_api_server.py]: I changed this part, is it working now?

File: fastchat/serve/openai_api_server.py (hunk 2 above)
Comment (soulsyrup) [reply fastchat/serve/openai_api_server.py]: I've added the import and also the api/v1/

**Pull request merged**